src/app/services/constants.ts

```diff
@@ -893,3 +894,13 @@ export const CLOZE_ITEM_SECTION_ID = "non-selected-items";
 export const CLOZE_DROP_ZONE_ID_PREFIX = "drop-zone-";
 // Matches: [drop-zone], [drop-zone|w-50], [drop-zone|h-50] or [drop-zone|w-50h-200]
 export const dropZoneRegex = /\[drop-zone(?<params>\|(?<index>i-\d+?)?(?<width>w-\d+?)?(?<height>h-\d+?)?)?]/g;
+
+export const USER_ROLES: Role[] = [
```

Comment:
These are the same values as in the `Role` type. There may be a way in TypeScript to generate this without having to hardcode them.

Reply:
It looks like this is not possible, because you can't convert a union type (which is a compile-time constant) into an array with those strings in it (which is a run-time value).